### PR TITLE
Updated helm chart to render same yaml for STS as in Kyma

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- For name `eventing-name`, the label `app.kubernetes.io/instance` should have value `eventing` instead of `eventing-nats`, so that the upgrade of StatefulSet works.

**Related issue(s)**

- https://github.com/kyma-project/nats-manager/issues/33

### CODE REVIEW

File: resources/nats/templates/statefulset.yaml [hunk 1]
```diff
@@ -144,6 +144,7 @@ spec:
         - -routez
         - -subz
         - -varz
+        - -healthz
```
Comment (mfaizanse) [inline resources/nats/templates/statefulset.yaml:147]: Synced from [PR](https://github.com/kyma-project/kyma/pull/17392)

File: config/samples/operator_v1alpha1_nats.yaml
Comment (k15r) [inline config/samples/operator_v1alpha1_nats.yaml:1]: why do we delete this file in this PR?

Comment (mfaizanse) [reply config/samples/operator_v1alpha1_nats.yaml:1]: This is a duplicate of this: https://github.com/kyma-project/nats-manager/blob/255437ed555d6308a02a892ff9aac0dc5fe23de7/config/samples/eventing-nats-prod.yaml

**Review by k15r — APPROVED** [review]

**Pull request merged**